runelite-client/src/main/java/net/runelite/client/plugins/poh/BurnerOverlay.java

```diff
@@ -66,42 +61,83 @@ public Dimension render(Graphics2D graphics)
 			return null;
 		}
 
+		// Get each object in POH
 		plugin.getPohObjects().forEach((object, tile) ->
 		{
+			// If planes match
 			if (tile.getPlane() == client.getPlane())
 			{
-				if (BURNER_UNLIT.contains(object.getId()))
+				if (!plugin.getBURNER_LIT().contains(object.getId()))
 				{
-					drawBurner(graphics, "Unlit", object, Color.RED);
+					return;
 				}
-				else if (BURNER_LIT.contains(object.getId()))
+
+				if (config.showBurner() && plugin.getCountdownTimerMap().containsKey(tile))
+				{
+					DecimalFormat decimalFormat = new DecimalFormat("#.##");
+					double certainSec = Double.valueOf(decimalFormat.format(plugin.getCountdownTimerMap().get(tile)));
+					double randomSec = Double.valueOf(decimalFormat.format(plugin.getRandomTimerMap().get(tile)));
+					ProgressPieComponent pieComponent = new ProgressPieComponent();
+					pieComponent.setPosition(object.getCanvasLocation());
+					pieComponent.setProgress(certainSec / plugin.getCountdownTimer());
+
+					if (certainSec > 0)
+					{
+						renderPie(graphics, Color.GREEN, Color.GREEN, pieComponent);
+					}
+
+					else
+					{
+						pieComponent.setProgress(randomSec / plugin.getRandomTimer());
+
+						if (randomSec > 0)
+						{
+							renderPie(graphics, Color.ORANGE, Color.ORANGE, pieComponent);
+						}
+					}
+				}
+
+				/*
+				if (plugin.getBURNER_LIT().contains(object.getId()))
 				{
-					drawBurner(graphics, "Lit", object, Color.GREEN);
+					if (config.showBurner())
+					{
+						DecimalFormat df = new DecimalFormat("#.##");
+						if (plugin.getCountdownTimerMap().containsKey(tile))
+						{
+							double certainSec = plugin.getCountdownTimerMap().get(tile);
+							double randomSec = plugin.getRandomTimerMap().get(tile);
+							certainSec = Double.valueOf(df.format(certainSec));
+							randomSec = Double.valueOf(df.format(randomSec));
+							ProgressPieComponent pieTimer = new ProgressPieComponent();
+							pieTimer.setPosition(object.getCanvasLocation());
+							pieTimer.setProgress((certainSec / plugin.getCountdownTimer()));
+
+							if (certainSec > 0)
+							{
+								renderPie(graphics, Color.GREEN, Color.GREEN, pieTimer);
+							}
+							else
+							{
+								pieTimer.setProgress((randomSec / plugin.getRandomTimer()));
+								if (randomSec > 0)
+								{
+									renderPie(graphics, Color.ORANGE, Color.ORANGE, pieTimer);
+								}
+							}
+						}
+					}
 				}
+				 */
 			}
 		});
 		return null;
 	}
 
-	private void drawBurner(Graphics2D graphics, String text, TileObject tileObject, Color color)
+	private void renderPie(Graphics2D graphics, Color fillColor, Color borderColor, ProgressPieComponent pieProgress)
```

Comment:
this method can be static, but also idk why this method even exists as it is doing nothing